test/test_model.py

```diff
@@ -47,50 +47,51 @@ def test_trivial_model():
 def test_average_model():
     """Check the implementation of the average DW model."""
 
+    data = np.ones((100, 100, 100, 6), dtype=float)
+
     gtab = np.array(
         [
+            [0, 0, 0, 0],
             [-0.31, 0.933, 0.785, 25],
             [0.25, 0.565, 0.21, 500],
             [-0.861, -0.464, 0.564, 1000],
             [0.307, -0.766, 0.677, 1000],
-            [0.736, 0.013, 0.774, 2000],
+            [0.736, 0.013, 0.774, 1300],
         ]
     )
 
-    gtab_w25 = gtab[1:, :]
-    gtab_1000 = gtab[2:3, :]
-    gtab_2000 = gtab[2:, :]
+    data *= gtab[:, -1]
 
     tmodel_mean = model.AverageDWModel(gtab=gtab, bias=False, stat="mean")
     tmodel_median = model.AverageDWModel(gtab=gtab, bias=False, stat="median")
     tmodel_1000 = model.AverageDWModel(
-        gtab=gtab, bias=False, th_high=1000, th_low=1000
+        gtab=gtab, bias=False, th_high=1000, th_low=900
     )
     tmodel_2000 = model.AverageDWModel(
-        gtab=gtab, bias=False, th_high=2000, th_low=1000
+        gtab=gtab, bias=False, th_high=2000, th_low=900, stat="mean",
     )
 
     # Verify that fit function returns nothing
-    assert tmodel_mean.fit() is None
+    assert tmodel_mean.fit(data[..., 1:], gtab=gtab[1:].T) is None
 
-    tmodel_median.fit()
-    tmodel_1000.fit()
-    tmodel_2000.fit()
+    tmodel_median.fit(data[..., 1:], gtab=gtab[1:].T)
+    tmodel_1000.fit(data[..., 1:], gtab=gtab[1:].T)
+    tmodel_2000.fit(data[..., 1:], gtab=gtab[1:].T)
 
     # Verify that the right statistics is applied and that the model discard b-values < 50
-    assert np.all(tmodel_mean.predict() == np.mean(gtab_w25[:, :2], axis=0))
-    assert np.all(tmodel_median.predict() == np.median(gtab_w25[:, :2], axis=0))
+    assert np.all(tmodel_mean.predict([0, 0, 0]) == 950)
```

Comment:
Isn't the input to `predict` supposed to be a gradient table? That is, a 4-by-n array? Why does it accept a 3-vector?